### PR TITLE
 ttfsample: init at 0.3.0

### PR DESCRIPTION
**Pull request by StefanSchroeder**
###### Description of changes

This is a new package, ttfsample, a cli tool which creates a PNG image for a TTF or OTF font, written in Golang.
This is the first package by  the owner and author.
Tools has no other homepage than github: https://github.com/StefanSchroeder/ttfsample 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [n.a.] (Package updates) Added a release notes entry if the change is major or breaking
  - [n.a. ] (Module updates) Added a release notes entry if the change is significant
  - [n.a.] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

@fricklerhandwerk

### CODE REVIEW

**Comment by fricklerhandwerk** [comment]
Sorry I can't review packages, there are too many notifications as it is.

**Comment by StefanSchroeder** [comment]
Fair enough. No problem. I didn't expect you to do so. I just wanted to let you know that I am submitting my first package. 😏

**Review by proofconstruction — COMMENTED** [review]
Adding yourself to the `maintainer-list.nix` should be a separate commit

File: pkgs/tools/misc/ttfsample/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{ buildGoModule , fetchFromGitHub , lib
+}:
```
Comment (proofconstruction) [inline pkgs/tools/misc/ttfsample/default.nix:2]:
```suggestion
{ lib
, buildGoModule
, fetchFromGitHub
}:
```

File: pkgs/tools/misc/ttfsample/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        };
+        vendorSha256 = "sha256-7kse3MYIj3Q0WL/fgR6TFmMGNQUSKuJM7+IqB83rwFY=";
+        meta = with lib; {
+                description = "Create images of fonts";
+                homepage = "https://github.com/StefanSchroeder/ttfsample";
+                license = licenses.mit;
+                maintainers = with maintainers; [ stefanschroeder ];
+                mainProgram = "ttfsample";
+        };
+}
```
Comment (proofconstruction) [inline pkgs/tools/misc/ttfsample/default.nix:22]:
```suggestion
buildGoModule rec {
  pname = "ttfsample";
  version = "0.3.0";

  src = fetchFromGitHub {
    owner = "StefanSchroeder";
    repo = "ttfsample";
    rev = "${version}";
    hash = "sha256-qB6n0xaBvjmBOOId4SLjc5Oi59DbtvLdSUw98AZcn0o=";
  };

  vendorSha256 = "sha256-7kse3MYIj3Q0WL/fgR6TFmMGNQUSKuJM7+IqB83rwFY=";

  meta = with lib; {
    description = "Create images of fonts";
    homepage = "https://github.com/StefanSchroeder/ttfsample";
    license = licenses.mit;
    maintainers = with maintainers; [ stefanschroeder ];
    mainProgram = "ttfsample";
  };
}
```

Fixed whitespace

**Review by proofconstruction — CHANGES_REQUESTED** [review]

**Comment by proofconstruction** [comment]
The package builds successfully on `x86-64-linux`, but the [`meta.platforms` attribute](https://nixos.org/manual/nixpkgs/stable/#var-meta-platforms) is not set, so Hydra might not build the binary. Recommend `meta.platforms = lib.platforms.linux;` or `meta.platforms = lib.platforms.unix;`, if someone can test on MacOS

**Review by proofconstruction — CHANGES_REQUESTED** [review]

**Comment by StefanSchroeder** [comment]
> Adding yourself to the `maintainer-list.nix` should be a separate commit

 @proofconstruction I don't know how to do that after the fact. Please advise.

**Comment by fricklerhandwerk** [comment]
Needs a rebase for a separate commit to add @stefanschroeder to maintainers.

**Comment by StefanSchroeder** [comment]
I have now rebased locally to split the addition of the maintainer into a separate commit (as suggested), but now I am getting  a conflict when I try to push to my GH repo. I need help finding the correct procedure to update the remote.
This scenario is not covered here: https://nixos.wiki/wiki/Nixpkgs/Create_and_debug_packages
There is this warning **Warning: only use 'git rebase' on your commits, which have not been pushed and nobody else is working with already!** But there are no instructions how to proceed if you indeed *have* pushed to remote.

**Comment by StefanSchroeder** [comment]
I think I performed the rebase now as suggested to split the commit for the maintainer-list.

@proofconstruction

**Pull request closed without merging**